### PR TITLE
Store actual umasks in state file

### PR DESCRIPTION
**Pull request by twpayne**
This is a work-in-progress fix for #1066, not ready for merging yet.

### CODE REVIEW

**Comment by fwSmit** [comment]
Are you sure that it's a umask thing? I didn't change it (as far as I know), but the bug still occurred.

**Comment by twpayne** [comment]
I'm pretty sure that this is a bug in chezmoi, and actually the same as #1066, #1091, and #1155.

The tl;dr is that there are actually two bugs:
1. Due to incorrect umask handling, chezmoi is storing the last-written permissions in its persistent state incorrectly, so it keeps thinking that a file has changed when actually it hasn't.
2. The diff code is displaying an empty diff when it should be displaying that difference in permissions that chezmoi thinks exists (but actually doesn't).

This PR isn't complete yet. It's missing:
1. A proper set of tests.
2. A migration for existing chezmoi users. If the above theory is correct, then chezmoi2 has stored permissions incorrectly for many users, and needs to fix itself.
